File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.cs
public class Solution {
    public int MaxProfitAssignment(int[] difficulty, int[] profit, int[] worker) {
        int n = difficulty.Length;
        List<Tuple<int, int>> jobs = new List<Tuple<int, int>>();
        
        for (int i = 0; i < n; i++) {
            jobs.Add(new Tuple<int, int>(difficulty[i], profit[i]));
        }
        
        jobs.Sort((a, b) => a.Item1.CompareTo(b.Item1));
        Array.Sort(worker);
        
        int maxProfit = 0, jobIndex = 0, maxJobProfit = 0;
        
        foreach (int ability in worker) {
            while (jobIndex < n && ability >= jobs[jobIndex].Item1) {
                maxJobProfit = Math.Max(maxJobProfit, jobs[jobIndex].Item2);
                jobIndex++;
            }
            maxProfit += maxJobProfit;
        }
        
        return maxProfit;
    }
}

File: 0826-most-profit-assigning-work/0826-most-profit-assigning-work.cpp
class Solution {
public:
    int maxProfitAssignment(vector<int>& difficulty, vector<int>& profit, vector<int>& worker) {
        int n = difficulty.size();
        vector<pair<int,int>>jobs(n);
        for(int i=0; i<n; i++)
        {
            jobs[i] = {difficulty[i], profit[i]};
        }
        sort(jobs.begin(), jobs.end());
        sort(worker.begin(), worker.end());
        
        int maxProfit = 0, maxJobProfit = 0, jobIndex = 0;
        for(auto ability : worker)
        {
            while(jobIndex<n && ability>=jobs[jobIndex].first)
            {
                maxJobProfit = max(maxJobProfit, jobs[jobIndex].second);
                jobIndex++;
            }
            maxProfit+=maxJobProfit;
        }
        return maxProfit;
    }
};